projects/core/src/site-context/facade/site-context-persistence.service.ts

```diff
@@ -0,0 +1,87 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { ConsentStatePersistenceService } from '../../state/services/consent-state-peristence.service';
+import {
+  getContextParameterDefault,
+  getContextParameterValues,
+} from '../config/context-config-utils';
+import { SiteContextConfig } from '../config/site-context-config';
+import {
+  CURRENCY_CONTEXT_ID,
+  LANGUAGE_CONTEXT_ID,
+} from '../providers/context-ids';
+import { CurrencyService } from './currency.service';
+import { LanguageService } from './language.service';
+
+@Injectable({ providedIn: 'root' })
+export class SiteContextPersistenceService implements OnDestroy {
+  protected subscription1 = new Subscription();
+  protected subscription2 = new Subscription();
+
+  constructor(
+    protected statePersistenceService: ConsentStatePersistenceService,
+    protected languageService: LanguageService,
+    protected currencyService: CurrencyService,
+    protected config: SiteContextConfig
+  ) {}
+
+  public initSync(): void {
+    this.initLanguageSync();
+    this.initCurrencySync();
+  }
+
+  protected initLanguageSync(): void {
```

Comment:
This isn't clear to me. Do we allow customers to not persist a context all based on configuration? Or change from localStorrage to cookie or session storage?

Reply:
mmm, I probably stretch the scope here. No need to do this now, but it would be great to have flexibility going forward. It might mean that we need another conversation on this topic.